program-runtime/src/loaded_programs.rs

```diff
@@ -262,6 +262,38 @@ pub struct LoadedPrograms {
     entries: HashMap<Pubkey, Vec<Arc<LoadedProgram>>>,
 }
 
+#[derive(Debug, Default)]
+pub struct LoadedProgramsForSlot {
```

Comment:
This structure is not valid for the entire slot but only one TX batch in it. Maybe it should be called "ForBatch" instead?